src/libs/actions/Session/index.js

```diff
@@ -316,7 +317,12 @@ function setPassword(password, validateCode, accountID) {
                 return;
             }
 
-            Onyx.merge(ONYXKEYS.ACCOUNT, {error: Localize.translateLocal('setPasswordPage.accountNotValidated')});
+            const login = lodashGet(response, 'data.email', null);
+            Onyx.merge(ONYXKEYS.ACCOUNT, {validated: false, validateCodeExpired: true, error: Localize.translateLocal('setPasswordPage.accountNotValidated')});
+            if (login) {
+                Onyx.merge(ONYXKEYS.CREDENTIALS, {login});
```

Comment:
Why are we setting `login` here?

Reply:
If I randomly create hit the URL, by default I won't have the email of the user in the store. So when we hit the URL, I set this email I receive in the API response as the login, which then gets used to resend the magic link.

Reference to the discussion: https://github.com/Expensify/App/pull/5150#issuecomment-915677035

Reply:
By "randomly create hit the URL" do you mean that we do not have the `login` stored from a previous session?

>by default I won't have the email of the user in the store
>I set this email I receive in the API response as the login, which then gets used to resend the magic link

Mind explaining this in a code comment? There is no way for anyone looking at this code to make the connection between triggering a new magic link and having a login saved in the store.

Reply:
Please do this.